### PR TITLE
Feature-detail of module and list modules of workflow

### PR DESCRIPTION
**Pull request by siddhant-basepair**
This PR adds a certain feature in List Modules and Pipeline Flow

list pipeline_modules -u 10
list pipeline_modules --uid 10

it will show all the modules for pipeline id 10

list module -u 447

this will give detail of a module

Dependency on PR -
basepair/webapp#1126

Related Ticket --
https://app.asana.com/0/1201368373933132/1201406835914429

### CODE REVIEW

File: bin/basepair [hunk 1]
```diff
@@ -739,13 +769,16 @@ def _list_item(bp, data_type=None, uids=None, is_json=False):
...
   if data_type not in list_type:
     eprint('List data type - {} currently not supported.'.format(data_type))
     return
 
   if not uids:
-    eprint('At least one uid required.')
+    if data_type == 'modules':
+      eprint('At least one workflow id required.')
+    else:
+      eprint('At least one uid required.')
```
Comment (Lalee10) [inline bin/basepair:781]: Remove command specific code

File: bin/basepair [hunk 2]
```diff
@@ -739,13 +769,16 @@ def _list_item(bp, data_type=None, uids=None, is_json=False):
     eprint('Invalid data type.')
     return
 
-  list_type = ['analysis', 'genome', 'sample']
+  list_type = ['analysis', 'genome', 'sample', 'module', 'modules']
```
Comment (Lalee10) [inline bin/basepair:772]: rename `modules` -> `workflow_modules`

**Review by Lalee10 — CHANGES_REQUESTED** [review]

File: basepair/api.py [hunk 3]
```diff
@@ -1267,7 +1267,7 @@ def print_data(self, data_type='', is_json=False, uid=None, project=None):
     # if it is a list
     if data_type in list_methods:
       method = list_methods.get(data_type)
-      if data_type == 'modules':
+      if data_type == 'pipeline_modules':
```
Comment (Lalee10) [inline basepair/api.py:1270]: Remove module specific code, we don't need it since we are using args.uid now

File: bin/basepair [hunk 4]
```diff
@@ -769,16 +758,17 @@ def _list_item(bp, data_type=None, uids=None, is_json=False):
...
   if not uids:
-    if data_type == 'modules':
-      eprint('At least one workflow id required.')
-    else:
-      eprint('At least one uid required.')
+    eprint('At least one uid required.')
+    return
+
+  if data_type == 'pipeline_modules' and len(uids) > 1:
+    eprint('Please provide only one pipeline uid.')
```
Comment (Lalee10) [inline bin/basepair:771]: Can we ignore other ids and use just the first one?

**Review by Lalee10 — CHANGES_REQUESTED** [review]

File: basepair/api.py (hunk 3 above)
Comment (siddhant-basepair) [reply basepair/api.py:1270]: we need it cause we need the uid as function parameter for this list method only

File: bin/basepair (hunk 4 above)
Comment (siddhant-basepair) [reply bin/basepair:771]: this is necessary as we need to restrict the input to one uid from client-side

**Review by Lalee10 — APPROVED** [review]

**Pull request merged**